packages/auto-drive/src/api/models/uploads.ts

```diff
@@ -74,17 +74,33 @@ export type CompleteUploadResponse = {
   cid: string
 }
 
-export type UploadFileStatus = {
-  type: 'file'
-  progress: number
-  cid?: string
-}
+export type UploadFileStatus =
+  | {
+      completed: true
+      type: 'file'
+      progress: number
+      cid: string
+    }
+  | {
+      completed: false
+      type: 'file'
+      progress: number
+      cid: null
+    }
 
-export type UploadFolderStatus = {
-  type: 'folder'
-  progress: number
-  cid?: string
-}
+export type UploadFolderStatus =
```

Comment:
Personally I find these two types a bit repetitive, why not making use of generic types?

```typescript
type UploadStatus<Type extends 'file' | 'folder'> = 
  | {
      completed: true
      type: Type
      progress: number
      cid: string
    }
  | {
      completed: false
      type: Type
      progress: number
      cid: null
    }

export type UploadFileStatus = UploadStatus<'file'>
export type UploadFolderStatus = UploadStatus<'folder'>
```

Reply:
Updated! Thanks